src/components/modals/LinkShare.js

```diff
@@ -31,13 +38,26 @@ const LinkShare = ({ cancel, mapId }) => {
         setStage("copied");
     }
 
-    return <>
-        <div className="modal-title">Export</div>
+    if (!mapToShare && currentMapId == null)
+        return <Modal id="link">
+            <div className="modal-title">Generate GeoJSON</div>
+            <div className="modal-content">
+                <div>Please save map first!</div>
+            </div>
+        </Modal>
+
+    const reset = () => {
+        setLinkText("Generate link...");
+        setStage("generate");
+    }
+
+    return <Modal id="link" customClose={reset}>
+        <div className="modal-title">Generate GeoJSON</div>
         <input type="text" disabled={true} value={linkText} className='link-text-input'></input>
         <p className='modal-warning'>Note: this will make your map data available to anyone with this link</p>
```

Comment:
ik this isn't a line that you've changed but just spotted that this warning could be clearer. I'm not sure what 'make your map data available' means. Does link sharing give write access or readonly access to the map? Depending on the answer, the warning could say 'anyone with the link will be able to <view / edit> your map', similar to what it says on Google docs

Reply:
@rogup @lin-d-hop if you go on the link they can see the data, they can't change it, just looks like a big json. agree the warning could be in more natural/regular language

Reply:
This is the last unresolved comment on this PR.

@lin-d-hop Do you thnk we should change the text to something like "Note: anyone with this link will be able to view your map's data" ? Or do you think it's fine as it is?

Reply:
Cool, I'll make this change in #172 since it's just one line and this PR has been merged (because of the complicated branching for the last release)